server/api/lib/trees/regional_tree_permit.js

```diff
@@ -0,0 +1,210 @@
+			return tp; //original one, not compact
+		}
+	}).filter(Boolean); // remove undefined values
+	//save only the new ones
+	try { //TODO promise all or knex save bulk
+		new_tree_permits.map(async tp => {
+			console.log(`saving new tree permit: ${tp.attributes[tpc.PERMIT_NUMBER]} with ${tp.attributes[tpc.NUMBER_OF_TREES]} ${tp.attributes[tpc.TREE_NAME]} trees.`);
+			await tp.save();
+		});
+	}
+	catch (err) {
+		console.log(err);
+	}
+	return new_tree_permits;
+}
+
+const parseTreesXLS = async (filename) => {
+	// hack
+	const sheetname = path.parse(filename).name.toLowerCase().includes('after') ? SHEET_AFTER : SHEET_BEFORE;
+	const workbook = xlsx.readFile(filename);
+	const sheet = workbook.Sheets[sheetname];
+	const sheet_json = xlsx.utils.sheet_to_json(sheet, { raw: false });
+	const treePermits = sheet_json.map(row => {
+		return new TreePermit(
+			{
+				[tpc.REGIONAL_OFFICE]: row['אזור'],
+				[tpc.PERMIT_NUMBER]: row['מספר רשיון'],
+				[tpc.ACTION]: row['פעולה'], // cutting , copying
+				[tpc.PERMIT_ISSUE_DATE]: row['תאריך הרשיון'],
+				[tpc.PERSON_REQUEST_NAME]: row['מבקש'],
+				[tpc.START_DATE]: row['מתאריך'],
+				[tpc.END_DATE]: row['עד תאריך'],
+				[tpc.LAST_DATE_TO_OBJECTION]: row['תאריך אחרון להגשת ערער'], // column might be missing from
+				[tpc.APPROVER_NAME]: row['שם מאשר'],
+				[tpc.APPROVER_TITLE]: row['תפיד מאשר'],
+				// Location
+				[tpc.PLACE]: row['מקום הפעולה'],
+				[tpc.STREET]: row['רחוב'],
+				[tpc.STREET_NUMBER]: row['מספר'],
+				[tpc.GUSH]: row['גוש'],
+				[tpc.HELKA]: row['חלקה'],
+				// Trees details
+				[tpc.TREE_NAME]: row['שם העץ'],
+				[tpc.TREE_KIND]: row['סוג העץ'],
+				[tpc.NUMBER_OF_TREES]: row['מספר עצים'],
+				[tpc.REASON_SHORT]: row['סיבה'],
+				[tpc.REASON_DETAILED]: row['פרטי הסיבה'],
+				[tpc.COMMENTS_IN_DOC]: row['הערות לעצים']
+			}
+		);
+	});
+	return treePermits;
+};
+
+function generateFilenameByTime(url) {
+	const parsedFile = path.parse(url);
+	const basename = parsedFile.name.toLowerCase() + '-' + moment().format('YYYY-MM-DD-hh-mm-ss') + parsedFile.ext.toLowerCase();
+	const filename = path.resolve(RAW_DATA_FOLDER, basename);
+	return filename;
+}
+
+async function crawlRegionalTreePermit(url) {
+	try {
+		const filename = generateFilenameByTime(url);
+		const treePermits = await getRegionalTreePermitsFromFile(url, filename);
+		const newTreePermits = await saveNewTreePermits(treePermits);
+		console.log('Extracted ' + newTreePermits.length + ' new permits from: ' + filename);
+		return newTreePermits.length;
+	} catch (err) {
+		console.log(err);
+		return false;
+	}
+}
+const regionalTreePermit = () => {
+	Promise.allSettled(regionalTreePermitUrls.map(url => crawlRegionalTreePermit(url)))
+		.then((results) => {
+			let sumNewPermits = 0;
+			results.forEach(element => {
+				sumNewPermits = sumNewPermits + element.value;
+			});
+			console.log(`Done! Total ${sumNewPermits} new permits`);
+		})
+		.catch(err => console.log(err))
+		.finally(() => process.exit());
```

Comment:
maybe `process.exit()` should be left for the caller to use?
it is relevant when running the standalone bin file, but if we run this function from other places we might not want to exit